### ISSUE TITLE
Mod Chest Feature

### ISSUE THREAD

**Issue by MoeXIV** [drtshock/HCFProblems#137]
Insiding is the #1 thing I've seen on HCF, and it causes so many players to leave and quit the server.
I thought it'd be a cool feature to make a single chest that when a sign placed on it [mod] or something of the sorts, it only allows mods and + to access the single chest. This will help with insiders, as well as the fact it's only a single chest worth of goodies can be stored.

;)

**Comment by Lilmac12312** [comment]
Basically this is something like HCTeams has called a "subclaim". Only those who are modded within the would be able to open the "subclaim" unless Moe was just wanting it so only you can open the subclaim you've made. If this was to be added it'd be nice to have an option for it to be enabled / disabled.

**Comment by MoeXIV** [comment]
Correct +1

**Comment by Doxify** [comment]
Would like to see this added as well.

Maybe a command as well?

/f modchest (while looking at the chest)

**Comment by LOOOOOOOL** [comment]
I dont want this add it so make it configurable.

**Comment by Lilmac12312** [comment]
@LOOOOOOOL Well isn't that funny. First you're complaing about how you want the /f eotw command added yet never thought about everyone. If you took the time and looked over the past comments on this idea or known as a new feature you would of seen I mentioned adding in a configured option of enabling or disabling the "mod chest".

Let the developers take their time into adding what we've suggested instead of being a bossy cunt like you were above "I dont want this add it so make it confiureable.", honestly quite rude don't you think? Go get some respect seriously, I find it pathetic how you think you can treat someone providing a service to you like that..

**Comment by LOOOOOOOL** [comment]
Get a license and a brain, then try to trash talk me you suck up. You suck
up to everyone. Prplz and all of them.

**Comment by nsporillo** [comment]
@MoeXIV Particle effects sound cool, similar to how ender chests have particles if possible would be neat. Good idea.

**Comment by MoeXIV** [comment]
What can I say, I'm full of ideas ;)

**Comment by MoeXIV** [comment]
Can't edit a post on an iPhone so I'll make another. Any ETA on this feature?

**Comment by Doxify** [comment]
Yes I'd like the particle effects, make it so once you run the /f modchest command it places a sign on the chest that cannot be broken by anyone.

I think you should also add an option to make the chest a regular chest again, so if you look at the chest and type /f modchest off it will get rid of the sign on the chest and make it a regular chest again.

I need this asap, so I'm whiling to pay the $40/ hour fee if this can be done in ~3 hours or less, and no I don't want this feature only for myself, everyone should have it. That's the non-selfish way.

**Comment by MoeXIV** [comment]
Thank you, Dox. That's a really nice way of looking at it. I needed it soon as well.

I hope this kicks in.

**Comment by nsporillo** [comment]
Well I need to plan out the following:

* The command (and everything related to it, TL, arguments, permissions, etc)
* The listener (to enforce the mod chest rules)
* The actual system to handle who has access to the mod chest

It is not something I can go in and add in an hour, atleast not without it being half assed / buggy.

As far as an ETA, keep in mind that I have a job outside HCF, I'm a student, and I have a life as well. With that said I still find I have plenty of time in a week to contribute to HCF. However, there are days where i'm just completely unavailable.

I'm going to try to add this tonight though. Will not commit anything that is not tested to the best of my ability. If any of you would like to test, contact me on Skype later tonight, it'd definitely help out.

**Comment by Doxify** [comment]
DM me on twitter and I'll get on to help :)

**Comment by LOOOOOOOL** [comment]
Make it confirguable as I don't like this nor do I like any of doxfies
ideas.

**Comment by nsporillo** [comment]
It's obviously configurable, just dont give your players the permission node to make modchests.

**Comment by ArchDaWiz** [comment]
@nsporillo Make this configurable. Subclaims are a good idea, but i dont want it on my server. I might enable later, but for now, I see it ruining the point of "hardcore factions".

**Comment by nsporillo** [comment]
Have gotten pretty far on this, but there is still *a lot* to work on. In fact, not sure if it's even a good idea to still continue with this. Revamping Roles (and the commands / TL that correspond) entail only a portion of the changes. Will likely be awhile before this can be implemented correctly :cry:

**Comment by MoeXIV** [comment]
If you do decide to implement this, make sure the chest cannot be emptied by hoppers.

I understand this is a hard thing to implement, but I assure you, this is a well wanted feature and I'm sure people will love it.

**Comment by nsporillo** [comment]
@MoeXIV thanks for the reminder, that slipped my mind. Still need to add lockette-esque signs or particle effects to the mod chests as well. It will be a great addition but it's a big update so it will take time to roll out correctly.

**Comment by MoeXIV** [comment]
Sounds good +1

**Comment by KriticalPvP** [comment]
Vouch +1

**Comment by MoeXIV** [comment]
Any ETA? I know it's going to take some time, and I understand. Just curious! c;

**Comment by oSaphyre** [comment]
Any update about this? Still waiting, if u need testers feel free to contact me throught skype: bionicleg0d

**Comment by Lotteries** [comment]
Make sure you can not /f modchest in another factions land.. That would be pretty funny... I can imagine how many people would be mad. Also, what if a faction member wants to pretty much 'inside' there faction. They can just /f modchest the chest with all the goodies and no one can get the goodies back! Add it so if the faction member gets kicked the 'Mod chest' gets unclaimed.

**Comment by MoeXIV** [comment]
@lotteries you make no sense. How if the member wants to inside the chest is locked? All faction admins and mods will be able to access it no matter what.

It should not be "un modchested" if he is kicked because it's still accessible...

**Comment by Doxify** [comment]
Not sure what Lotteries is talking about but just to make explain what I originally wanted.

Admins / Mods can /f modchest a chest and then all admins and mods can access that chest, if one gets kicked, the chest is still a mod chest.

**Comment by Lotteries** [comment]
@MoeXIV I edited the comment, should make more sense now, i was basically stating some things that needed to be added with this to make it as simple as possible.

**Comment by MoeXIV** [comment]
Looks good. I think you should be able to place hoppers nearby for reasons obvious (making what you need to).

Maybe disabling the moving of items through through hoppers if it's a mod chest

**Comment by Lotteries** [comment]
@MoeXIV Thats true, i've edited the comment once again. It's @nsporillo's choice which way he wants to prevent hoppers from being used on mod chests, but i think denying it from being placed next to it is the safest way because there is no possible way to glitch it (at least that i know of!)

**Comment by MoeXIV** [comment]
Then replace the hopper from like hopper:0 to like, hopper:1 and disable all movements from hoppers with :1 meta.

**Comment by MoeXIV** [comment]
It ruins the aspect of HCF where you are limited where to place blocks in your own claim

**Comment by Doxify** [comment]
I was able to test this with nick and it's nearly done, hoppers still moved items out when I tested it but he said he's fixing that so I don't think you guys should worry about not being to place certain blocks in your claimed land

**Comment by KriticalPvP** [comment]
@Doxify Thank You! :)

**Comment by MoeXIV** [comment]
Great!

**Comment by MoeXIV** [comment]
Bumping this so it doesn't close due o inactivity.

**Comment by nsporillo** [comment]
@MoeXIV this wont need to be bumped, it won't be closed. 

It is however a bit of an undertaking and while I have gotten close a few weeks ago, still needs work.
